app/controllers/ProductController.scala

```diff
@@ -35,13 +38,127 @@ class ProductController @Inject() (
   }
 
   /** When the user route to <root>/rest/bakery/products then this will return all
-   * the elements of the product database.
-   *
+   *  the elements of the product database.
    * @return
    */
   def getInventory: Action[AnyContent] = Action {
-    val results : List[Product] = getResults[Product]
+    val results : List[Products] = getResults[Products]
+
+    Ok(Json.toJson(Json.toJson(results)))
+  }
+
+  /** When the user GET to <root>/rest/bakery/product/:id where id is the UUID for a product,
+   * make sure it's a valid UUID string (or an exception will get thrown) then check if it found
+   * a result or not (result is empty) and return the appropriate status + body if needed.
+   * @param id
+   * @return
+   */
+  def getProduct(id: String): Action[AnyContent] = Action {
+
+    // Used variable isError to make sure I don't overload the method by giving multiple status
+    var isError = false
+    var result: List[Products] = List.empty
+
+    // Strings not being correct UUID format will throw an exception, so this will catch it
+    try
+      result = getResultById[Products](id)
+    catch {
+      case x: PSQLException =>
+        isError = true
+    }
+
+    // I have to make sure I don't keep giving multiple statuses so this if-else will resolve it
+    if (result.isEmpty && !isError)
+      NotFound("ID not found")
+    else if(isError)
+      NotFound("Wrong ID string")
+    else
+      Ok(Json.toJson(Json.toJson(result)))
+  }
+
+  /** When the user POST to <root>/rest/bakery/product with a JSON request then this
+   *  will insert the given JSON into the database.
+   * @return
+   */
+  def addProduct(): Action[JsValue] = Action(parse.json) { request =>
```

Comment:
so the interface to the user will be different from the data that you are storing. For example:

```
case class Products(id: String,
                    name: String,
                    quantity: Option[Int],
                    price: Option[Double],
                    createdAt: Option[String],
                    updatedAt: Option[String])
```
you will want to require a name, quantity and price in the request (doesn't make sense to store products that don't have that). But you don't want your user to set their own `ids`, `createdAt` or `updatedAt` values. Those should be controlled by your application

Reply:
I agree, but that is why the SQL query only takes in the name, quantity, and price. So in reality even if they add the other fields, it does nothing. On the other hand, they exist so those fields are shown whenever the user `GET` a product, it would need to show these fields from my understanding, so it is provided.

Reply:
A user should not get access to an interface that is not relevant to them. If fields are not needed, they should be removed. You can create a specific request model to handle this and map between the request and the actual database stored value. The current model can still be used through the GET method though (since it is information the user might want or find useful)

Reply:
Alright, I'll make another case class userProduct and make sure it only has the fields only accessable to a user.